File: frontend/src/app/dto/BaseApiDto.ts
export interface IBaseApiDto<Type> {
    status: string;
    data: Type[] | Type;
    message: string[] | undefined;
}

export class BaseApiDto<Type> implements IBaseApiDto<Type> {
    status!: string;
    data!: Type[] | Type;
    message: string[] | undefined;

    constructor(data?: IBaseApiDto<Type>) {
        if (data) {
            for (const property in data) {
                if (data.hasOwnProperty(property)) {
                    (<any>this)[property] = (<any>data)[property];
                }
            }
        }
    }
}

File: src/main/java/com/astrapay/dto/BaseApiDto.java
package com.astrapay.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BaseApiDto<T> {
    private String status;
    private List<String> message = new ArrayList<>();
    private T data;

    public BaseApiDto(T data) {
        this.data = data;
        this.status = "OK";
    }

    public BaseApiDto(T data, String message) {
        this.data = data;
        this.status = "OK";
        this.message.add(message);
    }

    public BaseApiDto(String status, List<String> message) {
        this.status = status;
        this.message = message;
    }
}